src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/impl/Channel.kt

```diff
@@ -47,37 +47,47 @@ import kotlin.concurrent.withLock
 
 
 class Channel(
-    private val address: InetSocketAddress,
-    secure: Boolean,
+    private val defaultAddress: InetSocketAddress,
+    private val defaultSecure: Boolean,
     private val sessionAlias: String,
     private val reconnectDelay: Long,
     private val handler: IProtocolHandler,
     private val mangler: IProtocolMangler,
     private val onEvent: (event: Event, parentEventId: EventID) -> Unit,
     private val onMessage: (RawMessage) -> Unit,
-    eventLoopGroup: EventLoopGroup,
-    private val actionExecutor: ActionStreamExecutor,
+    private val eventLoopGroup: EventLoopGroup,
+    sequencePool: TaskSequencePool,
     val parentEventId: EventID
 ) : IChannel, ITcpChannelHandler {
     private val logger = KotlinLogging.logger {}
     private val lock = ReentrantLock()
```

Comment:
Maybe use ReadWriteLock instead because the internal send operation is more frequent than open and close operations.

Reply:
Read lock can be acquired before releasing a write lock, but we must check chanal staus and open it if required under the write lock.